browser/net/url_context.h

```diff
@@ -27,6 +28,9 @@ struct BraveRequestInfo {
   uint64_t request_identifier = 0;
   size_t next_url_request_index = 0;
   net::HttpRequestHeaders* headers = nullptr;
+  const net::HttpResponseHeaders* original_response_headers = nullptr;
+  scoped_refptr<net::HttpResponseHeaders>* override_response_headers = nullptr;
+  GURL* allowed_unsafe_redirect_url = nullptr;
```

Comment:
Not needed for now, but just noting it might be cleaner to make a subclass with these extra members instead and cast as needed in the headers received code.

Reply:
Sounds good, I'll create a small follow-up issue to do this if that's fine with you.

Reply:
Just an issue in the backog is fine for now.